metaconfig-typesafe-config/src/main/scala/metaconfig/typesafeconfig/TypesafeConfig2Class.scala

```diff
@@ -30,7 +30,7 @@ object TypesafeConfig2Class {
         case lst: ConfigList =>
           Conf.Lst(lst.listIterator().asScala.map(loop).toList)
         case _ =>
-          value.unwrapped() match {
+          value.unwrapped match {
```

Comment:
I personally don't mind this change, but I would expect a drop-in Scala replacement to not require this diff even if that means using `()` to call non-side-effecting methods

Reply:
I really wasn't sure what to do as the internal tests in Scala where inconsistent how it called Java with or without parens. So overall, I tried to make it more Scala-like. I left them on for obvious side-effecting 0-arity methods. Someone who was really bored could do an entire API review 😆 

If it really could have been a drop-in then I think Lightbend should have serious considered the PR - it is close but will probably change again for Scala 3 support etc. I also still have to change the package which will require import changes. ☹️

Reply:
I don't mind the parenthesis diff, metaconfig touches such a tiny surface of the API anyways.

>  internal tests in Scala where inconsistent how it called Java with or without parens.

with or without parens works when calling Java methods, but Scala "nullary methods" have additional information in their signatures to prohibit calling 

> will probably change again for Scala 3 support

I doubt that ;) I suspect it will work without any changes.

This PR is very close to being ready!

- change package name so it doesn't conflict with `com.typesafe.config`
- new module `metaconfig-sconfig` to avoid conflict with the old one
- we can copy-paste the `metaconfig-typesafe-config` tests for sconfig, I hope to remove the typesafe-config tests at some point so duplication is no problem